h/search/query.py

```diff
@@ -15,6 +17,48 @@
 OFFSET_MAX = 9800
 
 
+def wildcard_uri_is_valid(wildcard_uri):
+    """
+    Return True if uri contains wildcards in appropriate places, return False otherwise.
+
+    Wildcards are not permitted in the scheme or netloc of the uri.
+    """
+    if "*" not in wildcard_uri and "?" not in wildcard_uri:
+        return False
+    # If uri is in the form of a unique id, the domain will not be parsed correctly but
+    # it is still valid. Check for this special case and return early.
+    if wildcard_uri.startswith("urn:x-pdf:"):
```

Comment:
Should this logic be applied to all non-HTTP URIs? ie. Should this be `if scheme not in ("http", "https")`?